pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -116,6 +116,18 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 		return reconcile.Result{}, fmt.Errorf("errors: %v %v", err, updateErr)
 	}
 
+	if buildRun.Labels == nil {
+		buildRun.Labels = make(map[string]string)
+	}
+	if buildRun.Labels[buildv1alpha1.LabelBuild] == "" {
```

Comment:
im not sure what is this doing, can you explain? I dont see this documented on your PR comments

Reply:
Add Build label in buildRun,
When deleting Build, we will query all BuildRun which is under this Build by this label.

This is not used or show for end user, we don't need to document it. So for our internal logic.

Reply:
I meant in your PR description, helpful for people like me reviewing. Ok, I see. So I think your labels need some cleaning, in a BuildRun they read like:
```
  labels:
    build.build.dev/generation: "1"
    build.build.dev/name: buildah-golang-build
```
I would have eexpected:

```
  labels:
    buildrun.build.dev/build-generation: "1"
    buildrun.build.dev/build-name: buildah-golang-build
```

Reply:
I think it is fine to use them in buildrun, just tell the name and generation of build:
```
  labels:
    build.build.dev/generation: "1"
    build.build.dev/name: buildah-golang-build
```
So that the end user can know what is it clearly and we can just use one shard name.

Because we also add labels in taskrun, like this:
```
buildv1alpha1.LabelBuild:              build.Name,
buildv1alpha1.LabelBuildGeneration:    strconv.FormatInt(build.Generation, 10),
buildv1alpha1.LabelBuildRun:           buildRun.Name,
buildv1alpha1.LabelBuildRunGeneration: strconv.FormatInt(buildRun.Generation, 10),
```

If we change like that, we also need to change to:
```
  labels:
    taskrun.build.dev/build-generation: "1"
    taskrun.build.dev/build-name: buildah-golang-build
    taskrun.build.dev/build-run-generation: "1"
    taskrun.build.dev/build--run-name: buildah-golang-buildrun
```

Then we need four new labels and names for generated taskrun, but I think the label name can be shared, we don't need to have those duplicate label names